modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/ClasspathBasedDslContext.java

```diff
@@ -0,0 +1,91 @@
+import static org.reflections.util.ClasspathHelper.forJavaClassPath;
+import org.mule.runtime.core.registry.SpiServiceRegistry;
+import org.mule.runtime.extension.api.annotation.Extension;
+import org.mule.runtime.extension.api.introspection.ExtensionFactory;
+import org.mule.runtime.extension.api.introspection.ExtensionModel;
+import org.mule.runtime.extension.api.introspection.declaration.DescribingContext;
+import org.mule.runtime.extension.api.introspection.declaration.spi.Describer;
+import org.mule.runtime.extension.xml.dsl.api.resolver.DslResolvingContext;
+import org.mule.runtime.module.extension.internal.DefaultDescribingContext;
+import org.mule.runtime.module.extension.internal.introspection.DefaultExtensionFactory;
+import org.mule.runtime.module.extension.internal.introspection.describer.AnnotationsBasedDescriber;
+import org.mule.runtime.module.extension.internal.introspection.version.StaticVersionResolver;
+
+import java.net.URL;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.reflections.Reflections;
+import org.reflections.scanners.SubTypesScanner;
+import org.reflections.scanners.TypeAnnotationsScanner;
+import org.reflections.util.ConfigurationBuilder;
+
+/**
+ * Implementation of {@link DslResolvingContext} that
+ * scans the Classpath looking for the {@link Class} associated to the
+ * provided Extension {@code name}
+ *
+ * @since 4.0
+ */
+class ClasspathBasedDslContext implements DslResolvingContext {
+
+  private final ClassLoader classLoader;
+  private final Map<String, Class<?>> extensionsByName = new HashMap<>();
+  private final Map<String, ExtensionModel> resolvedModels = new HashMap<>();
+  private final ExtensionFactory extensionFactory =
+      new DefaultExtensionFactory(new SpiServiceRegistry(), getClass().getClassLoader());
+
+  ClasspathBasedDslContext(ClassLoader classLoader) {
+    this.classLoader = classLoader;
+    findExtensionsInClasspath();
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Optional<ExtensionModel> getExtension(String name) {
+
+    if (!resolvedModels.containsKey(name) && extensionsByName.containsKey(name)) {
+      Describer describer = new AnnotationsBasedDescriber(extensionsByName.get(name), new StaticVersionResolver("4.0"));
+      DescribingContext context = new DefaultDescribingContext(getClass().getClassLoader());
+      resolvedModels.put(name, extensionFactory.createFrom(describer.describe(context), context));
+    }
+    return Optional.ofNullable(resolvedModels.get(name));
+  }
+
+
+  private void findExtensionsInClasspath() {
+
+    // default classpath for testing environment, were the extension classloader is not useful
+    Set<Class<?>> annotated = getExtensionTypes(forJavaClassPath());
+
+    if (annotated.isEmpty()) {
+      // we are building an standalone extension, thus the classloader of the extension is required
```

Comment:
i would prefer to do this search first and do the testing one in a second phase. But let me know if this is not possible.